modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModuleModel.java

```diff
@@ -137,6 +147,89 @@ public ArtifactAst expand() {
   }
 
   private ArtifactAst expand(List<ComponentAst> moduleComponentModels, Set<String> moduleGlobalElementsNames) {
+    if (existOperationThatUsesImplicitConfiguration() && hasXmlSdkPropertiesWithDefaultValues()) {
+
+      ComponentAst componentAst = new BaseComponentAst() {
+
+        @Override
+        public ComponentIdentifier getIdentifier() {
+          return ComponentIdentifier.builder()
+              .namespaceUri(extensionModel.getXmlDslModel().getNamespace())
+              .namespace(extensionModel.getXmlDslModel().getPrefix())
+              .name(IMPLICIT_CONFIG_NAME).build();
+        }
+
+        @Override
+        public ComponentType getComponentType() {
+          return CONFIG;
+        }
+
+        @Override
+        public ComponentLocation getLocation() {
+          return from(IMPLICIT_CONFIG_NAME);
```

Comment:
Would there be any issue between implicit configs of different xml sdk modules for having the same location/componentId?

Reply:
The app has only one implicit config if there are properties with default values

Reply:
If the app uses two xml module that can have implicit configs, there might be two.

Reply:
But there will be one config in the extension model. I think that is how it's designed now. I'll check again...

Reply:
I have already addressed that now